### PR TITLE
Explicitly require VS2015 or greater

### PR DESCRIPTION
**Pull request by jslee02**
Related issue: #184

### CODE REVIEW

File: CMakeModules/CompilerSettings.cmake [hunk 1]
```diff
@@ -27,12 +27,15 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
...
 # Visual Studio
-if(MSVC OR MSVC90 OR MSVC10)
+if(MSVC)
+    if(MSVC_VERSION VERSION_LESS 1900)
+        message(FATAL_ERROR "${PROJECT_NAME} requires Visual Studio 2015 or greater.")
+    endif()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc /MP /W1 /bigobj")
     if(FCL_TREAT_WARNINGS_AS_ERRORS)
         add_definitions(/WX)
     endif(FCL_TREAT_WARNINGS_AS_ERRORS)
```
Comment (sherm1) [inline CMakeModules/CompilerSettings.cmake]: Consider `endif()` for consistency.

Comment (jslee02) [inline CMakeModules/CompilerSettings.cmake]: Actually, I don't know which one is the convention here. Some `endif()` contains parameter while some don't.

Comment (sherm1) [inline CMakeModules/CompilerSettings.cmake]: I agree! `endif()` is always better, IMO. That is very clear when the original condition is some complicated expression!

**Comment by jslee02** [comment]
I removed conditions from `endif()`. It seems `endif()` is the convention according to the `CMakeLists.txt` files of FCL.

**Pull request merged**